Code/autopkglib/Unarchiver.py

```diff
@@ -85,10 +84,14 @@ def main(self):
             raise ProcessorError(
                 "Expected an 'archive_path' input variable but none is set!"
             )
-        destination_path = self.env.get(
-            "destination_path",
-            os.path.join(self.env["RECIPE_CACHE_DIR"], self.env["NAME"]),
-        )
+        destination_path = self.env.get("destination_path")
+        if not destination_path and not self.env.get("NAME"):
+            raise ProcessorError(
+                "No destination path provided and NAME is empty, "
+                "cannot pick a default unpack directory!"
+            )
+        elif not destination_path:
+            os.path.join(self.env["RECIPE_CACHE_DIR"], self.env["NAME"])
```

Comment:
Why not as a last resort, use `os.path.join(self.env["RECIPE_CACHE_DIR"], "unpack")` as the unpack directory?